vmdb/app/helpers/api_helper/renderer.rb

```diff
@@ -198,6 +198,7 @@ def collection_search(is_subcollection, type, klass)
         :userid         => @auth_user,
         :results_format => :objects
       }
+      options[:order] = sort_options              if sort_options.present? && res.respond_to?(:reorder)
```

Comment:
One option is to set a flag when this condition is true, then test that instead in both cases.